### PR TITLE
[Android] NDK19 support (needs build-tools 27.0.3)

### PR DESCRIPTION
**Pull request by peak3d**
## Description
NDK19 support

## Motivation and Context
NDK19 is current stable version and may provide fixes.

## How Has This Been Tested?
Compile kodi on Linux x86_64 host

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by peak3d** [comment]
jenkins build this please

File: tools/android/packaging/Makefile.in [hunk 1]
```diff
@@ -44,10 +44,12 @@ else ifeq ($(findstring arm,$(CPU)),arm)
 endif
 
 # libc++
-STLLIB=$(TOOLCHAIN)/$(HOST)/lib/libc++_shared.so
-# https://android.googlesource.com/platform/ndk/+/fe0fe5dc32682d1811d32492c3055becb512c147/build/tools/make_standalone_toolchain.py#308
```
Comment (Rechi) [inline tools/android/packaging/Makefile.in]: Why remove this comment? This is still correct for armeabi-v7a and NDK 18.

File: tools/android/packaging/Makefile.in [hunk 2]
```diff
@@ -44,10 +44,12 @@ else ifeq ($(findstring arm,$(CPU)),arm)
...
-ifeq ($(findstring armeabi-v7a,$(CPU)),armeabi-v7a)
+ifneq (,$(wildcard $(TOOLCHAIN)/sysroot/usr/lib/$(HOST)/libc++_shared.so))
+  STLLIB=$(TOOLCHAIN)/sysroot/usr/lib/$(HOST)/libc++_shared.so
+else ifeq ($(findstring armeabi-v7a,$(CPU)),armeabi-v7a)
   STLLIB=$(TOOLCHAIN)/$(HOST)/lib/armv7-a/libc++_shared.so
+else
+  STLLIB=$(TOOLCHAIN)/$(HOST)/lib/libc++_shared.s
```
Comment (Rechi) [inline tools/android/packaging/Makefile.in]:
```suggestion
  STLLIB=$(TOOLCHAIN)/$(HOST)/lib/libc++_shared.so
```

File: tools/depends/target/Toolchain.cmake.in [hunk 3]
```diff
@@ -73,6 +73,7 @@ set(CMAKE_FIND_ROOT_PATH @prefix@/@deps_dir@)
 set(CMAKE_LIBRARY_PATH @prefix@/@deps_dir@/lib)
 if(NOT "@use_toolchain@" STREQUAL "")
   list(APPEND CMAKE_FIND_ROOT_PATH @use_toolchain@ @use_toolchain@/@use_host@ @use_toolchain@/@use_host@/sysroot @use_toolchain@/@use_host@/sysroot/usr @use_toolchain@/@use_host@/libc @use_toolchain@/lib/@use_host@/sysroot @use_toolchain@/usr @use_toolchain@/sysroot/usr)
+  list(APPEND CMAKE_SYSTEM_LIBRARY_PATH @use_toolchain@/sysroot/usr/lib/@use_host@/@use_ndk_api@)
```
Comment (Rechi) [inline tools/depends/target/Toolchain.cmake.in]: This is only true for android, but this command is executed on all platforms.

File: tools/depends/target/Toolchain_binaddons.cmake.in [hunk 4]
```diff
@@ -64,6 +64,7 @@ endif()
 # where is the target environment
 if(NOT "@use_toolchain@" STREQUAL "")
   list(APPEND CMAKE_FIND_ROOT_PATH @use_toolchain@/sysroot/usr)
+  list(APPEND CMAKE_SYSTEM_LIBRARY_PATH @use_toolchain@/sysroot/usr/lib/@use_host@/@use_ndk_api@)
```
Comment (Rechi) [inline tools/depends/target/Toolchain_binaddons.cmake.in]: This is only true for android, but this command is executed on all platforms.

**Review by Rechi — CHANGES_REQUESTED** [review]
Either change the commit message and PR title to what it really does or add the missing changes for real NDK 19 support.

**Comment by peak3d** [comment]
@rechi the commit message tells what it does, it should still work with NDK 18, but there should be support for NDK 19 now

**Comment by Rechi** [comment]
Should or you actually did a complete successful depends rebuild with NDK 19 already?
Here target python fails to build with this PR and NDK 19.

**Pull request closed without merging**